d2l-image.js

```diff
@@ -1,11 +1,33 @@
-import '@polymer/polymer/polymer-legacy.js';
-import '@polymer/iron-ajax/iron-ajax.js';
-import { Polymer } from '@polymer/polymer/lib/legacy/polymer-fn.js';
-const $_documentContainer = document.createElement('template');
-
-$_documentContainer.innerHTML = `<dom-module id="d2l-image">
-	<template strip-whitespace="">
-		<style>
+import 'd2l-fetch/d2l-fetch.js';
+import { css, html, LitElement } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+
+class D2LImage extends LitElement {
+	static get properties() {
+		return {
+			/**
+			 * Alternate text describing the image
+			 * @type {string}
+			 */
+			altText: { type: String, attribute: 'alt-text' },
+			/**
+			 * URL of the image
+			 * @type {string}
+			 */
+			imageUrl: { type: String, attribute: 'image-url' },
+			/**
+			 * token used to authenticate the image retrieval
+			 * @type {string}
+			 */
+			token: { type: String },
+			_headers: { type: String, state: true },
+			_href: { type: String, state: true },
+			_imageUrl: { type: String, state: true }
```

Comment:
You don't need a `type` for [internal reactive state](https://lit.dev/docs/components/properties/#internal-reactive-state). Also I don't think `_headers` is used anymore?

Reply:
Oh interesting, I saw this pattern in some other code, but forgot about it. neat!

And yeah, I haven't cleaned everything up yet, I just copy/pasted the old properties 👍